### PR TITLE
infra: set renovate schedule to weekly

### PR DESCRIPTION
**Pull request by xDivisionByZerox**
This PR changes the interval from "at any time" to a weekly basis every Monday morning via a [cron](https://crontab.guru/crontab.5.html) expression. 

Important reference links:
- [Renovate schedule config](https://docs.renovatebot.com/configuration-options/#schedule)
- [Cron expression descriptor (for cron expression validation)](https://crontab.cronhub.io/)

### CODE REVIEW

File: .github/renovate.json5 [hunk 1]
```diff
@@ -3,6 +3,11 @@
   "extends": ["config:base", "schedule:daily", "group:allNonMajor"],
   "labels": ["c: dependencies"],
   "reviewersFromCodeOwners": true,
+  "schedule": [
+    // We must use the * wildcard for minutes, as Renovate doesn't support minute granularity
+    // Every minute, between 12:00 AM and 12:59 AM, only on Monday (minutes are not suppoted)
```
Comment (xDivisionByZerox) [inline .github/renovate.json5:8]: I just wanted to leave this here so it's clear why the expression is the way it is. Is this a valid thought or should this be removed?

**Review by Shinigami92 — DISMISSED** [review]
Maybe we should use just: https://docs.renovatebot.com/presets-schedule/#scheduleweekly

File: .github/renovate.json5 (hunk 1 above)
Comment (ST-DDT) [reply .github/renovate.json5:8]: Why every minute?

I also have found this, which might be a bit more readable although I don't know how this could be changed to achieve exactly what we want:

` "after 5pm on the 2nd day of the month"`

Comment (xDivisionByZerox) [reply .github/renovate.json5:8]: We want weekly, not monthly (as your phrase suggests).
Weekly would be `"at 12am on Monday"`.

Comment (xDivisionByZerox) [reply .github/renovate.json5:8]: > Why every minute?

Line 7. Renovate requires a '*' (translates to `every`) for the minute position.

**Comment by xDivisionByZerox** [comment]
> Maybe we should use just: https://docs.renovatebot.com/presets-schedule/#scheduleweekly

I don't see why. Cron expressions are a known standard for developers (at least to some degree). This language thing seems nice for readability purposes but has no documentation whatsoever. I also really dislike the idea of expressing what I want in plain English and a machine gets what I want.

I'll not block this, because if this. I just don't see the reason why this would be a required change for a merge on this PR.

**Comment by Shinigami92** [comment]
Everyone using renovate is using this syntax
Its the absolutely normal way of configuring renovate

- https://github.com/vitejs/vite/blob/5df788dfe2d89e541461e166f03afb38c2f1dd7e/.github/renovate.json5#L2
- https://github.com/prettier/plugin-pug/blob/d520345569c50dfa3350588bb835dce6ccf4e008/.github/renovate.json5#L3
- https://github.com/vitest-dev/vitest/blob/88d5764894f62449540209567df19926e46a91f3/.github/renovate.json5#L2

And what do you mean by not documented? I linked you the documentation ^^
You can even read more about it here: https://docs.renovatebot.com/config-presets/#how-to-use-preset-configs
And even more: https://docs.renovatebot.com/configuration-options/#schedule

**Comment by xDivisionByZerox** [comment]
So what does the syntax translates to when I write the following

>Take yourself 3 out times before midnight but only on a full moon in winter

Does this run at all? That's what I mean by documented. You linked just a bunch of examples that are marked as "documentation". If I can just write plain English, what words do actually get picked up for the expression?

But whatever, I will just change the expression.

File: .github/renovate.json5 [hunk 2]
```diff
@@ -3,6 +3,7 @@
   "extends": ["config:base", "schedule:daily", "group:allNonMajor"],
```
Comment (Shinigami92) [inline .github/renovate.json5:3]:
```suggestion
  "extends": ["config:base", "schedule:earlyMondays", "group:allNonMajor"],
```

File: .github/renovate.json5 [hunk 3]
```diff
@@ -3,6 +3,7 @@
...
   "labels": ["c: dependencies"],
   "reviewersFromCodeOwners": true,
+  "schedule": ["at 12am on Monday"],
```
Comment (Shinigami92) [inline .github/renovate.json5:6]:
```suggestion
```

**Review by Shinigami92 — CHANGES_REQUESTED** [review]

**Review by Shinigami92 — APPROVED** [review]

**Review by ST-DDT — APPROVED** [review]

**Comment by xDivisionByZerox** [comment]
If your have such specific standards to workflow configuration or code standards in general please don't ask next time in the meeting who whats to take the task if you insist on doing it your way :)

**Comment by Shinigami92** [comment]
> If your have such specific standards to workflow configuration or code standards in general please don't ask next time in the meeting who whats to take the task if you insist on doing it your way :)

But then I couldn't approve it and would need to ask another reviewer :see_no_evil:

**Comment by 1LISA2** [comment]
images (2)
下载 (2)
